src/commands/gate/evaluate.ts

```diff
@@ -119,22 +126,21 @@ export class GateEvaluateCommand extends Command {
   }
 
   private async evaluateRules(api: APIHelper, evaluateRequest: Payload): Promise<number> {
-    if (this.shouldWait()) {
-      this.context.stdout.write(renderWaiting())
-      await this.delay(this.waitingTime)
-    }
-
     this.context.stdout.write(renderGateEvaluationInput(evaluateRequest))
 
-    return retryRequest(
-      () => api.evaluateGateRules(evaluateRequest, this.context.stdout.write.bind(this.context.stdout)),
-      {
-        onRetry: (e, attempt) => {
+    // retry is handled in evaluateRulesWithWait, so we set maxTimeout and minTimeout as 0 for retryRequest
```

Comment:
this is going to be confusing so let's use the chance to be clearer in the comment
```suggestion
    /** 
    * `retryRequest` does not allow setting a wait time dependent on a backend response,
    * so we handle the wait time in `evaluateRulesWithWait`: we'll wait whatever is necessary 
    * for the returned promise to resolve or be rejected. The retry will start immediately after 
    * and will be handled by `retryRequest`.
    */
```